src/main/java/commons/serializers/ProducerRecordSerializer.java

```diff
@@ -0,0 +1,44 @@
+package commons.serializers;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import commons.headers.Headers;
+import producer.ProducerRecord;
+
+public class ProducerRecordSerializer<K,V> extends Serializer<ProducerRecord<K,V>> {
+    private final Class<K> keyClass;
+    private final Class<V> valueClass;
+
+    public ProducerRecordSerializer(Class<K> keyClass, Class<V> valueClass) {
+        this.keyClass = keyClass;
+        this.valueClass = valueClass;
+    }
+
+    @Override
+    public void write(Kryo kryo, Output output, ProducerRecord<K, V> record) {
+        output.writeString(record.getTopic());
+        output.writeLong(record.getTimestamp());
+
+        // Optional fields
+        kryo.writeObjectOrNull(output, record.getPartitionNumber(), Integer.class);
+        kryo.writeObjectOrNull(output, record.getKey(), keyClass);
+
+        kryo.writeObject(output, record.getValue());
+        kryo.writeObject(output, record.getHeaders());
+    }
+
+    @Override
+    public ProducerRecord<K, V> read(Kryo kryo, Input input, Class<? extends ProducerRecord<K, V>> aClass) {
+        String topic = input.readString();
+        Long timestamp = input.readLong();
+        Integer partitionNumber = kryo.readObjectOrNull(input, Integer.class);
+        K key = kryo.readObjectOrNull(input, keyClass);
+        V value = kryo.readObject(input, valueClass);
```

Comment:
OOC, why isn't this `V value = kyro.readObjectOrNull`?

Reply:
`value` is required for each record

Reply:
I double-checked the class and it seems like there's nothing preventing the value from being `null`: https://github.com/tferdous17/flux/blob/main/src/main/java/producer/ProducerRecord.java

Reply:
>is required for each record

Yep required in the sense that it's not an `Optional` type, but the value could still be null

Reply:
Are you suggesting, I should make `ProducerRecord<K,V>` mandate the type of `value`?

Reply:
I think you can update that line to use `readObjectOrNull`. I'm not sure about the constraints of the flux producer record, but if the value is supposed to be non-null, then i think mandating could makes sense. But that would be great for a separate PR

Reply:
ya the value for any record should be non-null as it contains the contents of the record. Key is different as kafka allows it to be optional which we also do in flux